sdk/storage/AzureStorageBlob/Source/Options/ClientMethodOptions.swift

```diff
@@ -72,27 +76,198 @@ public struct ListContainersOptions: RequestOptions {
+    /// only if both of the following conditions are met:
+    /// - The container's lease is currently active.
+    /// - The specified lease ID matches that of the container.
+    public let leaseAccessConditions: LeaseAccessConditions?
+
+    /// Initialize an `GetConainerOptions` structure.
+    /// - Parameters:
+    ///   - clientRequestId: A client-generated, opaque value with 1KB character limit that is recorded in analytics logs.
+    ///   - cancellationToken: A token used to make a best-effort attempt at canceling a request.
+    ///   - dispatchQueue: A dispatch queue on which to call the completion handler. Defaults to `DispatchQueue.main`.
+    ///   - context: A `PipelineContext` object to associate with the request.
+    ///   - timeout: Request timeout in seconds.
+    ///   - leaseAccessConditions: Options for accessing a blob based on the condition of a lease. If specified, the
+    ///     operation will be performed only if both of the following conditions are met:
+    ///     - The blob's lease is currently active.
+    ///     - The specified lease ID matches that of the blob.
+    public init(
+        clientRequestId: String? = nil,
+        cancellationToken: CancellationToken? = nil,
+        dispatchQueue: DispatchQueue? = nil,
+        context: PipelineContext? = nil,
+        timeout: TimeInterval? = nil,
+        leaseAccessConditions: LeaseAccessConditions? = nil
+    ) {
+        self.clientRequestId = clientRequestId
+        self.cancellationToken = cancellationToken
+        self.dispatchQueue = dispatchQueue
+        self.context = context
+        self.timeout = timeout
+        self.leaseAccessConditions = leaseAccessConditions
+    }
+}
+
+public struct CreateContainerOptions: RequestOptions {
+    /// A client-generated, opaque value with 1KB character limit that is recorded in analytics logs.
+    public let clientRequestId: String?
+
+    /// A token used to make a best-effort attempt at canceling a request.
+    public let cancellationToken: CancellationToken?
+
+    /// A dispatch queue on which to call the completion handler. Defaults to `DispatchQueue.main`.
+    public let dispatchQueue: DispatchQueue?
+
+    /// A `PipelineContext` object to associate with the request.
+    public var context: PipelineContext?
+
+    /// Request timeout in seconds.
+    public let timeout: TimeInterval?
+
+    /// Specifies a user-defined name-value pair associated with the blob. If no name-value pairs are specified, the operation will copy the metadata from the
+    /// source blob or file to the destination blob. If one or more name-value pairs are specified, the destination blob is created with the specified metadata, and
+    /// metadata is not copied from the source blob or file. Note that beginning with version 2009-09-19, metadata names must adhere to the naming rules for
+    /// C# identifiers. See Naming and Referencing Containers, Blobs, and Metadata for more information.
```

Comment:
> C# identifiers

Is this true because the service is written in C# and requires this? Or is this a mistake?

Reply:
No idea. It was copied from their service documentation.

Reply:
Hm, okay, I would bet that this is a limitation of the fact that the service is written in C# and thus applies to all clients regardless of language.